src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.FruitOperationDao;
+import core.basesyntax.model.FruitOperation;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private FruitOperationDao fruitOperationDao;
+    private OperationStrategy strategy;
+
+    public ShopServiceImpl(FruitOperationDao fruitOperationDao, OperationStrategy strategy) {
+        this.fruitOperationDao = fruitOperationDao;
+        this.strategy = strategy;
+    }
+
+    @Override
+    public void changeQuantityStore(List<FruitOperation> fruits) {
+        for (FruitOperation fruit : fruits) {
+            if (fruit.getOperation() == FruitOperation.Operation.BALANCE) {
```

Comment:
Avoid using switch-cases and if-else constructions for operation handling. It is recommended to use the Strategy pattern instead. You can pass the strategy map into the service that chooses the correct strategy for each operation. This will help you to adhere to the Open/Closed principle, making it easier to add new operations in the future without modifying the existing code.